apollo-runtime/src/main/java/com/apollographql/apollo/internal/ApolloCallTracker.java

```diff
@@ -0,0 +1,174 @@
+package com.apollographql.apollo.internal;
+
+import com.apollographql.apollo.ApolloCall;
+import com.apollographql.apollo.ApolloPrefetch;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * ApolloCallTracker is responsible for keeping track of running {@link ApolloCall} & {@link ApolloPrefetch} objects.
+ */
+public class ApolloCallTracker {
+
+  private Runnable idleCallback;
+
+  private final Deque<ApolloCall> runningSyncCalls = new ArrayDeque<>();
```

Comment:
I think yeah, a `LinkedHashSet` makes more sense here and would give us better performance over a `Deque`